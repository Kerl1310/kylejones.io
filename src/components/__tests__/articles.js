import React from 'react';
import renderer from 'react-test-renderer';
import { StaticQuery } from 'gatsby';

import Articles from '../articles';

beforeEach(() => {
  StaticQuery.mockImplementationOnce(({ render }) =>
    render({
      allMarkdownRemark: {
        edges: [
          {
            node: {
              frontmatter: {
                path: './my-path',
                date: '01-01-2001',
                title: 'my-title',
                isFeatured: true,
                featuredImage: './my-image.png',
              },
            },
          },
          {
            node: {
              frontmatter: {
                path: './my-second-path',
                date: '02-02-2002',
                title: 'my-second-title',
                isFeatured: false,
                featuredImage: './my-second-image.jpg',
              },
            },
          },
        ],
      },
    })
  );
});

describe('Articles', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Articles />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
