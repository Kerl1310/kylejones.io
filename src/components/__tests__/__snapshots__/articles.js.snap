// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Articles renders correctly 1`] = `
<div
  className="recent-articles-container"
>
  <h2
    className="recent-articles-title"
  >
    Recent Articles
  </h2>
  <div
    className="sc-bdVaJa iDvsqb recent-articles-link-group-container"
  >
     
    <div
      className="sc-bwzfXH hLZZVN recent-articles-link-container"
      id="recent-articles-link-container-/my-path"
    >
      <a
        className="recent-articles-link"
        href="/my-path"
        id="recent-articles-link-/my-path"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="__blank"
      >
        <strong>
          my-title
        </strong>
      </a>
    </div>
    <div
      className="sc-bwzfXH hLZZVN recent-articles-link-container"
      id="recent-articles-link-container-/my-second-path"
    >
      <a
        className="recent-articles-link"
        href="/my-second-path"
        id="recent-articles-link-/my-second-path"
        onClick={[Function]}
        onMouseEnter={[Function]}
        rel="noopener noreferrer"
        target="__blank"
      >
        <strong>
          my-second-title
        </strong>
      </a>
    </div>
  </div>
</div>
`;
